docs/dev_101_services.md

```diff
@@ -136,3 +136,62 @@ If you need more control over the service call, you can also pass an async funct
 async def custom_set_sleep_timer(entity, service_call):
     await entity.set_sleep_timer(service_call.data['sleep_time'])
 ```
+
+## Response Data
+
+Services may optionally respond to a service call with data for powering more advanced automations. The use of response data for services is meant for
+cases not a fit for the Home Assistant state e.g. a response stream of repeated objects, or API would be better queried on demand or filtered such as a search.
```

Comment:
```suggestion
cases that do not fit the Home Assistant state. For example, a response stream of repeated objects, or API would be better queried on demand or filtered such as a search.
```
I don't quite understand this sentence. What's the link between Home Assistant state and querying or filtering?

Reply:
Here is what i'm kind of trying to say: People try to stick stuff in a normal entity, but its like way too complicated or changes a lot. Say you have a schedule of all upcoming events -- that is too much data to put into a sensor attribute. Usually sensors just have a single value or some small number of attributes.  That data may be a good fit for a service response data.

Conversely, don't use service response data when you should use an entity.


The "filtered" part is not worded well. Let me take that out.

Reply:
not sure I can follow just yet. 
basically, what you are saying: if you want to transmit a large amount of data, such as a response stream of objects or an API, stick it into the response data, not into some other attribute?
If so, then I still don't understand how the querying, filtering, and search comes in.